internal/component/redis/redis.go

```diff
@@ -72,12 +75,32 @@ type RedisClient interface {
 	TTLResult(ctx context.Context, key string) (time.Duration, error)
 }
 
-func ParseClientFromProperties(properties map[string]string, defaultSettings *Settings) (client RedisClient, settings *Settings, err error) {
-	if defaultSettings == nil {
-		settings = &Settings{}
-	} else {
-		settings = defaultSettings
+func ParseClientFromProperties(properties map[string]string, componentType metadata.ComponentType) (client RedisClient, settings *Settings, err error) {
+	settings = &Settings{}
+
+	// upgrade legacy metadata properties
+	if componentType == metadata.StateStoreType || componentType == metadata.LockStoreType {
+		// Apply legacy defaults
+		settings.RedisMaxRetries = 3
+		settings.RedisMinRetryInterval = Duration(2 * time.Second)
+		// Parse legacy keys
+		if _, ok := properties["redisMinRetryInterval"]; !ok {
```

Comment:
```suggestion
		if properties["redisMinRetryInterval"] == "" {
```

Reply:
In here and below, present isn't relevant as much as it is being empty or not empty.